File: BackEnd/src/main/java/com/example/pifinance_back/Entities/ProjetInvestissement.java
package com.example.pifinance_back.Entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;

import javax.persistence.*;
import java.io.Serializable;
import java.time.LocalDate;
import java.util.Set;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class ProjetInvestissement implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    private String nom;
    private String description;
    private LocalDate date_debut;
    private LocalDate date_fin;
    private Double cout_initial;
    private float taux_interet;
    private String localisation;
    private String responsable_projet;
    @Enumerated(EnumType.STRING)
    private StatutProjet statutProjet;
    @OneToMany(mappedBy="projetInvestissement",cascade = CascadeType.ALL)
    @JsonIgnore
    private Set<FluxTresorerie> fluxTresoreries;
}

File: BackOffice/src/app/Models/ProjetInvestissement.ts
export enum StatutProjet {
    ENCOURS = "ENCOURS",
    ANNULE = "ANNULE",
    ENATTENTE = "ENATTENTE",
    TERMINE="TERMINE"
  }

export class ProjetInvestissement {
    id: any;
    nom: any;
    description:any;
    date_debut:any;
    date_fin:any;
    cout_initial:any;
    localisation:any;
    responsable_projet:any; 
    taux_interet: any;
    statutProjet:StatutProjet;

}
